src/langchain_google_cloud_sql_pg/postgresql_engine.py

```diff
@@ -116,24 +125,40 @@ async def _create(
         region: str,
         instance: str,
         database: str,
+        user: Optional[str] = None,
+        password: Optional[str] = None,
         loop: Optional[asyncio.AbstractEventLoop] = None,
         thread: Optional[Thread] = None,
     ) -> PostgreSQLEngine:
-        credentials, _ = google.auth.default(
-            scopes=["https://www.googleapis.com/auth/userinfo.email"]
-        )
-        iam_database_user = await _get_iam_principal_email(credentials)
+        if bool(user) ^ bool(password):
+            raise ValueError(
+                "Only one of 'user' or 'password' were specified. Either "
+                "both should be specified to use basic user/password "
+                "authentication or neither for IAM DB authentication."
+            )
         if cls._connector is None:
             cls._connector = await create_async_connector()
+        # if user and password are given, use basic auth
+        if user and password:
+            enable_iam_auth = False
+            db_user = user
+        # otherwise use automatic IAM database authentication
+        else:
+            # get application default credentials
+            credentials, _ = google.auth.default(
+                scopes=["https://www.googleapis.com/auth/userinfo.email"]
+            )
+            db_user = await _get_iam_principal_email(credentials)
+            enable_iam_auth = True
 
         # anonymous function to be used for SQLAlchemy 'creator' argument
         async def getconn() -> asyncpg.Connection:
             conn = await cls._connector.connect_async(  # type: ignore
                 f"{project_id}:{region}:{instance}",
                 "asyncpg",
-                user=iam_database_user,
+                user=db_user,
```

Comment:
forgot to pass password, having it set to `None` will work here as that is what we do for IAM authn internally anyways: https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/blob/main/google/cloud/sql/connector/asyncpg.py#L53

```suggestion
                user=db_user,
                password=password,
```